lib/App/index.js

```diff
@@ -365,6 +365,39 @@ class App {
             });
           }
         }
+
+        if (driver.connectivity && driver.connectivity.includes('matter')) {
+          if (!driver.matter) {
+            throw new Error(`drivers.${driver.id} invalid 'connectivity': 'matter' requires a 'matter' object.`);
+          }
+
+          if (driver.pair) {
+            throw new Error(`drivers.${driver.id} invalid 'pair': matter drivers do not support custom pairing views.`);
+          }
+
+          if (
+            (driver.matter.deviceVendorId === undefined && driver.matter.deviceProductName !== undefined) ||
+            (driver.matter.deviceVendorId !== undefined && driver.matter.deviceProductName === undefined)
+          ) {
+            throw new Error(`drivers.${driver.id} invalid 'matter': 'deviceVendorId' and 'deviceProductName' must be defined together.`);
+          }
+
+          const deviceJsExists = await this._fileExistsCaseSensitive(join('drivers', driver.id, 'device.js'));
+          
+          if (deviceJsExists) {
+            throw new Error(`drivers.${driver.id}: using a device.js file is not supported for Matter drivers.`);
+          }
+
+          const driverJsExists = await this._fileExistsCaseSensitive(join('drivers', driver.id, 'driver.js'));
```

Comment:
Ik denk dat dit niet lekker gaat werken we hebben sowieso al .ts en straks ook .mjs. Misschien maar gewoon niet doen deze error?